From 639661fdb5e4b7cc7a46cf64d5eacbc52fa9e721 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Mon, 19 Oct 2015 08:08:08 +0200
Subject: [PATCH 1/2] __pkg_commit_info() * check for FrugalBuild first , if
 found auto generate $commitmsg * if not print just exit and print a nice msg
 * without $commitmsg is like : ------$arch

---
 repoman | 22 ++++++++++++++--------
 1 file changed, 14 insertions(+), 8 deletions(-)

diff --git a/repoman b/repoman
index 76f9dff..a6e4092 100755
--- a/repoman
+++ b/repoman
@@ -1,22 +1,22 @@
 #!/bin/bash
 #
 #   repoman, a script to build and maintain packages
-#  
+#
 #   Copyright (c) 2005, 2006, 2007, 2008, 2009, 2010, 2011 by Miklos Vajna <vmiklos@frugalware.org>
-#  
+#
 #   This program is free software; you can redistribute it and/or modify
 #   it under the terms of the GNU General Public License as published by
 #   the Free Software Foundation; either version 2 of the License, or
 #   (at your option) any later version.
-# 
+#
 #   This program is distributed in the hope that it will be useful,
 #   but WITHOUT ANY WARRANTY; without even the implied warranty of
 #   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 #   GNU General Public License for more details.
-#  
+#
 #   You should have received a copy of the GNU General Public License
 #   along with this program; if not, write to the Free Software
-#   Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, 
+#   Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
 #   USA.
 #
 
@@ -601,9 +601,15 @@ $@" | darcs rec -a --pipe .
 __pkg_commit_info()
 {
 	local commitmsg
-	commitmsg="$pkgname-$pkgver-$pkgrel-$arch"
-	msg "Your git commit command is:"
-	msg "git commit -a -e -m $commitmsg"
+	if [ -e ./FrugalBuild ]; then
+		commitmsg="$pkgname-$pkgver-$pkgrel-$arch"
+		msg "Your git commit command is:"
+		msg "git commit -a -e -m $commitmsg"
+	else
+		msg "Cannot find any FrugalBuild in `pwd`"
+		msg "Commit command cannot be auto generated"
+		exit 1
+	fi
 }
 
 commit_info()
-- 
2.6.2

